core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+  }
+
+  @Test
+  public void createViewThatAlreadyExistsAsTable() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    TableIdentifier tableIdentifier = TableIdentifier.of("ns", "table");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(tableIdentifier.namespace());
+    }
+
+    assertThat(tableCatalog().tableExists(tableIdentifier)).as("Table should not exist").isFalse();
+
+    tableCatalog().buildTable(tableIdentifier, SCHEMA).create();
+
+    assertThat(tableCatalog().tableExists(tableIdentifier)).as("Table should exist").isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(tableIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withDefaultNamespace(tableIdentifier.namespace())
+                    .withQuery("spark", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("Table with same name already exists: ns.table");
+  }
+
+  @Test
+  public void createTableThatAlreadyExistsAsView() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).as("View should not exist").isFalse();
+
+    catalog()
+        .buildView(viewIdentifier)
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(viewIdentifier.namespace())
+        .withQuery("spark", "select * from ns.tbl")
+        .create();
+
+    assertThat(catalog().viewExists(viewIdentifier)).as("View should exist").isTrue();
+
+    assertThatThrownBy(() -> tableCatalog().buildTable(viewIdentifier, SCHEMA).create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View with same name already exists: ns.view");
+  }
+
+  @Test
+  public void renameView() {
+    TableIdentifier from = TableIdentifier.of("ns", "view");
+    TableIdentifier to = TableIdentifier.of("ns", "renamedView");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(from.namespace());
+    }
+
+    assertThat(catalog().viewExists(from)).as("View should not exist").isFalse();
+
+    catalog()
+        .buildView(from)
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(from.namespace())
+        .withQuery("spark", "select * from ns.tbl")
+        .create();
+
+    assertThat(catalog().viewExists(from)).as("View should exist").isTrue();
+
+    catalog().renameView(from, to);
+
+    assertThat(catalog().viewExists(from)).as("View should not exist with old name").isFalse();
+    assertThat(catalog().viewExists(to)).as("View should exist with new name").isTrue();
```

Comment:
Minor: Is it possible to reorder the method calls and call `assertThat(...).isFalse().as(...)` instead? I think that would be easier to read because you wouldn't need to skip the message to find out whether the assertion matches your assumption about the behavior.

Reply:
unfortunately no, because the `.as()` part needs to come before the final assertion. Otherwise it's going to be ignored